aea/decision_maker/messages/base.py

```diff
@@ -23,11 +23,13 @@
 from copy import copy
 from typing import Any, Dict, Optional
 
+from aea.configurations.base import PublicId
+
 
 class InternalMessage:
     """This class implements a message."""
 
-    protocol_id = "internal"
+    protocol_id = PublicId("fetchai", "internal", "0.1.0")
```

Comment:
not sure about this. Should the internal protocol use a public ID? The answer is yes if we want the internal protocol be treated as any other protocol (e.g. stored in the protocol registry). However, it is a very special one. Maybe it is safe to do this since there will not be any other internal protocol different than `fetchai/internal` (so no risk of name conflicts).

Reply:
Yes I thought about this too. Since eventually we want an actor or actor-like architecture, it makes sense to give all messages the same structure. That way it will be trivial to send them between internal components/actors.

Reply:
Since there is little cost, i.e. only maintaining no naming conflict with `fetchai/internal` I think its ok